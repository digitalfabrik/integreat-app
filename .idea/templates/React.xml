<templateSet group="React">
  <template name="rrc" value="import React, {ReactElement} from 'react';&#10;import { connect } from 'react-redux';&#10;&#10;type $TM_FILENAME_BASE$Props = {&#10;}&#10;type $TM_FILENAME_BASE$State = {&#10;}&#10;class $TM_FILENAME_BASE$ extends React.Component&lt;$TM_FILENAME_BASE$Props,$TM_FILENAME_BASE$State&gt; {&#10; constructor(props: $TM_FILENAME_BASE$Props) {&#10;  super(props);&#10;&#10;  this.state = {};&#10;&#10; }&#10;&#10; render() {&#10;  return (&#10;   &lt;div&gt;&#10;    $END$&#10;   &lt;/div&gt;&#10;  );&#10; }&#10;}&#10; &#10;const mapStateToProps: (state: IState) =&gt; MapStateToProps = state =&gt; ({&#10; &#10;});&#10;    &#10;export default connect(&#10;    mapStateToProps,&#10;)($componentName$);&#10;" description="Creates a React component class connected to redux" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="" defaultValue="" alwaysStopAt="true" />
    <variable name="componentName" expression="" defaultValue="&quot;MyComponent&quot;" alwaysStopAt="true" />
    <context>
      <option name="TypeScript" value="true" />
    </context>
  </template>
  <template name="rcfc" value="import React, { Component, ReactElement } from 'react';&#10;&#10;type $TM_FILENAME_BASE$Props = {&#10;}&#10;type $TM_FILENAME_BASE$State = {&#10;}&#10;&#10;class $componentName$ extends Component&lt;$TM_FILENAME_BASE$Props,$TM_FILENAME_BASE$State&gt; {&#10;    constructor(props: $TM_FILENAME_BASE$Props) {&#10;        super(props);&#10;    }&#10;    &#10;    componentWillMount() {&#10;    &#10;    }&#10;    &#10;    componentDidMount() {&#10;    &#10;    }&#10;    &#10;    componentWillReceiveProps(nextProps) {&#10;    &#10;    }&#10;    &#10;    shouldComponentUpdate(nextProps, nextState) {&#10;    &#10;    }&#10;    &#10;    componentWillUpdate(nextProps, nextState) {&#10;    &#10;    }&#10;    &#10;    componentDidUpdate(prevProps, prevState) {&#10;    &#10;    }&#10;    &#10;    componentWillUnmount() {&#10;    &#10;    }&#10;    &#10;    render(): ReactElement {&#10;        return (&#10;            &lt;div&gt;&#10;                $END$&#10;            &lt;/div&gt;&#10;        );&#10;    }&#10;}&#10;&#10;export default $componentName$;&#10;" description="Creates a React component class with all lifecycle methods and ES6 module system" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="" defaultValue="" alwaysStopAt="true" />
    <variable name="componentName" expression="" defaultValue="&quot;MyComponent&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="rfc" value="import React, {ReactElement} from 'react';&#10;&#10;type $TM_FILENAME_BASE$Props = {&#10;}&#10;&#10;const $TM_FILENAME_BASE$: React.FC&lt;$TM_FILENAME_BASE$Props&gt; = (props:$TM_FILENAME_BASE$Props):ReactElement =&gt; {&#10;    return (&#10;        &lt;div&gt;&#10;            $END$&#10;        &lt;/div&gt;&#10;    );&#10;};&#10;&#10;export default $TM_FILENAME_BASE$;&#10;" description="Creates a stateless React component with interface ES6 module system" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
  <template name="con" value="constructor(props) {&#10;    super(props: $TM_FILENAME_BASE$Props);&#10;    $END$&#10;}&#10;" description="Adds a default constructor for the class that contains props as arguments" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="conc" value="constructor(props, context) {&#10;    super(props, context);&#10;    $END$&#10;}&#10;" description="Adds a default constructor for the class that contains props and context as arguments" toReformat="true" toShortenFQNames="true">
    <context />
  </template>
  <template name="cwm" value="componentWillMount() {&#10;    $END$&#10;}&#10;" description="Invoked once, both on the client and server, immediately before the initial rendering occurs" toReformat="true" toShortenFQNames="true">
    <context />
  </template>
  <template name="cdm" value="&#10;componentDidMount() {&#10;    $END$&#10;}&#10;" description="Invoked once, only on the client (not on the server), immediately after the initial rendering occurs." toReformat="true" toShortenFQNames="true">
    <context />
  </template>
  <template name="scu" value="&#10;shouldComponentUpdate(nextProps, nextState) {&#10;    $END$&#10;}&#10;" description="Invoked before rendering when new props or state are being received." toReformat="true" toShortenFQNames="true">
    <context>
      <option name="JS_EXPRESSION" value="true" />
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="cwup" value="componentWillUpdate(nextProps, nextState) {&#10;    $END$&#10;}&#10;" description="Invoked immediately before rendering when new props or state are being received." toReformat="true" toShortenFQNames="true">
    <context />
  </template>
  <template name="cdup" value="&#10;componentDidUpdate(prevProps, prevState) {&#10;    $END$&#10;}&#10;" description="Invoked immediately after the component's updates are flushed to the DOM." toReformat="true" toShortenFQNames="true">
    <context />
  </template>
  <template name="cwun" value="componentWillUnmount() {&#10;    $END$&#10;}&#10;" description="Invoked immediately before a component is unmounted from the DOM." toReformat="true" toShortenFQNames="true">
    <context />
  </template>
  <template name="ren" value="&#10;render(): ReactElement {&#10;    return (&#10;        &lt;div&gt;&#10;            $END$&#10;        &lt;/div&gt;&#10;    );&#10;}&#10;" description="When called, it should examine this.props and this.state and return a single child element." toReformat="true" toShortenFQNames="true">
    <context />
  </template>
  <template name="ssf" value="&#10;this.setState((state, props) =&gt; { &#10;    return {&#10;        $END$ &#10;    };&#10;});&#10;" description="Performs a shallow merge of nextState into current state" toReformat="false" toShortenFQNames="true">
    <context />
  </template>
  <template name="rcc" value="import React, { Component } from 'react';&#10;&#10;class $TM_FILENAME_BASE$ extends Component {&#10; render() {&#10;  return (&#10;   &lt;div&gt;&#10;    $END$&#10;   &lt;/div&gt;&#10;  );&#10; }&#10;}&#10;&#10;export default $TM_FILENAME_BASE$;" description="Creates a A higher-order component" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="rpc" value="import React, {ReactElement} from 'react';&#10;&#10;type $TM_FILENAME_BASE$Props = {&#10;}&#10;type $TM_FILENAME_BASE$State = {&#10;}&#10;class $TM_FILENAME_BASE$ extends React.PureComponent&lt;$TM_FILENAME_BASE$Props,$TM_FILENAME_BASE$State&gt; {&#10; constructor(props: $TM_FILENAME_BASE$Props) {&#10;  super(props);&#10;&#10;  this.state = {};&#10;&#10; }&#10;&#10; render(): ReactElement {&#10;  return (&#10;   &lt;div&gt;&#10;    $END$&#10;   &lt;/div&gt;&#10;  );&#10; }&#10;}&#10;&#10;export default $TM_FILENAME_BASE$;" description="Creates a React pure component class with PropTypes and ES6 module system" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="rrdc" value="import React, { Component } from 'react';&#10;import { connect } from 'react-redux';&#10;&#10;type MapStateToProps = {&#10;}&#10;type MapDispatchToProps = {&#10;}&#10;&#10;type $TM_FILENAME_BASE$Props = MapStateToProps &amp; MapDispatchToProps &amp; {&#10;}&#10;type $TM_FILENAME_BASE$State = {&#10;}&#10;&#10;&#10;class $TM_FILENAME_BASE$ extends React.Component&lt;$TM_FILENAME_BASE$Props ,$TM_FILENAME_BASE$State&gt; {&#10; render() {&#10;  return (&#10;   &lt;div&gt;&#10;    $END$&#10;   &lt;/div&gt;&#10;  );&#10; }&#10;}&#10;&#10;const mapStateToProps: (state: IState) =&gt; MapStateToProps = state =&gt; ({&#10; &#10;});&#10;&#10;const mapDispatchToProps : (dispatch: Dispatch&lt;Action&gt;) =&gt; MapDispatchToProps = dispatch  =&gt; {&#10; return {&#10; }&#10;}&#10;&#10;export default connect(&#10; mapStateToProps, mapDispatchToProps&#10;)($TM_FILENAME_BASE$);" description="Creates a React component class connected to redux with dispatch" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="rccc" value="import React, {ReactElement} from 'react';&#10;&#10;type $TM_FILENAME_BASE$Props = {&#10;}&#10;type $TM_FILENAME_BASE$State = {&#10;}&#10;&#10;class $TM_FILENAME_BASE$ extends React.Component&lt;$TM_FILENAME_BASE$Props,$TM_FILENAME_BASE$State&gt; {&#10; constructor(props: $TM_FILENAME_BASE$Props) {&#10;  super(props);&#10;&#10;  this.state = {};&#10;&#10; }&#10;&#10; render(): ReactElement {&#10;  return (&#10;   &lt;div&gt;&#10;    $END$&#10;   &lt;/div&gt;&#10;  );&#10; }&#10;}&#10;&#10;export default $TM_FILENAME_BASE$;" description="Creates a React component class with constructor and interfaces" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
  <template name="useState" value="const [$state$, set$setState$] = useState&lt;&gt;($initState$);$END$" description="useState" toReformat="true" toShortenFQNames="true">
    <variable name="state" expression="" defaultValue="&quot;state&quot;" alwaysStopAt="true" />
    <variable name="setState" expression="capitalize(state)" defaultValue="" alwaysStopAt="true" />
    <variable name="initState" expression="" defaultValue="&quot;initState&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="useEffect" value="useEffect(() =&gt; {&#10;$effect$&#10;  return () =&gt; {&#10;  // CleanUp&#10;    $effect$&#10;  };&#10;}, [$deps$]);&#10;$END$" description="useEffect" toReformat="true" toShortenFQNames="true">
    <variable name="effect" expression="" defaultValue="&quot;effect&quot;" alwaysStopAt="true" />
    <variable name="deps" expression="" defaultValue="&quot;deps&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="useCallback" value="const memoizedCallback = useCallback(&#10;  () =&gt; {&#10;    $callback$&#10;  },&#10;  [$input$],&#10;);&#10;$END$" description="useCallback" toReformat="true" toShortenFQNames="true">
    <variable name="callback" expression="" defaultValue="&quot;doSomething(a, b)&quot;" alwaysStopAt="true" />
    <variable name="input" expression="" defaultValue="&quot;a, b&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="useMemo" value="const memoizedValue = useMemo(() =&gt; computeExpensiveValue(a, b), [a, b]);$END$" description="useMemo" toReformat="true" toShortenFQNames="true">
    <context />
  </template>
  <template name="rntc" value="import React from 'react'&#10;import { render } from '@testing-library/react-native'&#10;&#10;describe('$TM_FILENAME_BASE$', () =&gt; {&#10;  it('should render $TM_FILENAME_BASE$', () =&gt; {&#10;    const { getByText } = render(&lt;&gt;&lt;/&gt;)&#10;    expect(getByText('send')).toBeDisabled()&#10;  })&#10;})" description="Create standard component test with react-native testing-library" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rnfc" value="import React, {ReactElement} from 'react';&#10;import {View} from 'react-native'&#10;&#10;type $TM_FILENAME_BASE$Props = {&#10;}&#10;&#10;const $TM_FILENAME_BASE$: React.FC&lt;$TM_FILENAME_BASE$Props&gt; = (props:$TM_FILENAME_BASE$Props):ReactElement =&gt; {&#10;    return (&#10;        &lt;View&gt;&#10;            $END$&#10;        &lt;/View&gt;&#10;    );&#10;};&#10;&#10;export default $TM_FILENAME_BASE$;&#10;" description="Creates a stateless React-Native component with interface ES6 module system" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rnccc" value="import React, {ReactElement} from 'react';&#10;import {View} from 'react-native'&#10;&#10;type $TM_FILENAME_BASE$Props = {&#10;}&#10;type $TM_FILENAME_BASE$State = {&#10;}&#10;class $TM_FILENAME_BASE$ extends React.Component&lt;$TM_FILENAME_BASE$Props,$TM_FILENAME_BASE$State&gt; {&#10; constructor(props: $TM_FILENAME_BASE$Props) {&#10;  super(props);&#10;&#10;  this.state = {};&#10;&#10; }&#10;&#10; render(): ReactElement {&#10;  return (&#10;   &lt;View&gt;&#10;    $END$&#10;   &lt;/View&gt;&#10;  );&#10; }&#10;}&#10;&#10;export default $TM_FILENAME_BASE$;" description="Creates a React component class with constructor and interfaces" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
  <template name="scnv" value="const $componentName$ = styled.View`&#10;`" description="styled view (native component)" toReformat="false" toShortenFQNames="true">
    <variable name="componentName" expression="" defaultValue="&quot;StyledView&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
  <template name="scni" value="const $componentName$ = styled.Image`&#10;`" description="styled image(native component)" toReformat="true" toShortenFQNames="true">
    <variable name="componentName" expression="" defaultValue="&quot;StyledImage&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
  <template name="scnt" value="const $componentName$ = styled.Text`&#10;`" description="styled text (native component)" toReformat="true" toShortenFQNames="true">
    <variable name="componentName" expression="" defaultValue="&quot;StyledText&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
  <template name="useRef" value="const $ref$ = useRef&lt;&gt;($initialValue$);$END$" description="useRef" toReformat="true" toShortenFQNames="true">
    <variable name="ref" expression="" defaultValue="&quot;ref&quot;" alwaysStopAt="true" />
    <variable name="initialValue" expression="" defaultValue="&quot;initialValue&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="cref" value="this.$refName$Ref = React.createRef&lt;&gt;();$END$" description="Creates a ref" toReformat="true" toShortenFQNames="true">
    <variable name="refName" expression="" defaultValue="&quot;refName&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="hoc" value="import React from 'react'&#10;&#10;export default ($WrappedComponent$) =&gt; {&#10;  const $hocComponent$ = ({ ...props }) =&gt; &lt;$WrappedComponent$ {...props} /&gt;&#10;&#10;  return $hocComponent$&#10;}" description="Creates a React higher-order component" toReformat="true" toShortenFQNames="true">
    <variable name="WrappedComponent" expression="" defaultValue="&quot;WrappedComponent&quot;" alwaysStopAt="true" />
    <variable name="hocComponent" expression="" defaultValue="&quot;hocComponent&quot;" alwaysStopAt="true" />
    <context />
  </template>
  <template name="disp" value="const mapDispatchToProps : (dispatch: Dispatch&lt;Action&gt;) =&gt; MapDispatchToProps = dispatch  =&gt; {&#10; return {&#10; }&#10;}" description="Adds the redux mapDispatchToProps function" toReformat="true" toShortenFQNames="true">
    <context />
  </template>
  <template name="rnpc" value="import React, {ReactElement} from 'react';&#10;import {View} from 'react-native'&#10;&#10;type $TM_FILENAME_BASE$Props = {&#10;}&#10;type $TM_FILENAME_BASE$State = {&#10;}&#10;&#10;class $TM_FILENAME_BASE$ extends React.PureComponent&lt;$TM_FILENAME_BASE$Props,$TM_FILENAME_BASE$State&gt; {&#10; constructor(props: $TM_FILENAME_BASE$Props) {&#10;  super(props);&#10;&#10;  this.state = {};&#10;&#10; }&#10;&#10; render(): ReactElement {&#10;  return (&#10;   &lt;View&gt;&#10;    $END$&#10;   &lt;/View&gt;&#10;  );&#10; }&#10;}&#10;&#10;export default $TM_FILENAME_BASE$;" description="Creates a React pure component class with PropTypes and ES6 module system" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rtc" value="import React from 'react'&#10;import { render } from '@testing-library/react'&#10;&#10;describe('$TM_FILENAME_BASE$', () =&gt; {&#10;  it('should render $TM_FILENAME_BASE$', () =&gt; {&#10;    const { getByText } = render(&lt;&gt;&lt;/&gt;)&#10;    expect(getByText('send')).toBeDisabled()&#10;  })&#10;})" description="Create standard component test with react testing-library" toReformat="true" toShortenFQNames="true">
    <variable name="TM_FILENAME_BASE" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="&quot;&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_TOP_LEVEL_STATEMENT" value="true" />
    </context>
  </template>
</templateSet>