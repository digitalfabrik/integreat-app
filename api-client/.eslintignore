/flow-typed/
__coverage__
/node_modules/
