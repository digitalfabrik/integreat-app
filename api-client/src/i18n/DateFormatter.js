// @flow

import type Moment from 'moment'
import moment from 'moment'

export type FormatFunctionType = (date: Moment, options: {| format?: string |}) => string

class DateFormatter {
  _fallbackFormat: string | void
  _defaultLocale: string

  constructor (fallbackFormat: string | void, defaultLocale: string) {
    this._fallbackFormat = fallbackFormat
    this._defaultLocale = defaultLocale
  }

  format: FormatFunctionType = (date: Moment, options: {| format?: string |}) => {
    const format = options.format || this._fallbackFormat
    // TODO IGAPP-399: Uncomment again and use locale instead of hardcoded 'en'
    // const requestedLocale = defaultLocale
    const requestedLocale = 'en'
    // $FlowFixMe locales is not included in the flow types
    const allLocales = moment.locales()
    const locale = allLocales.includes(requestedLocale) ? requestedLocale : this.defaultLocale
    return date.locale(locale).format(format)
  }

  set defaultLocale (value: string) {
    this._defaultLocale = value
  }

  get defaultLocale (): string {
    return this._defaultLocale
  }
}

export default DateFormatter
