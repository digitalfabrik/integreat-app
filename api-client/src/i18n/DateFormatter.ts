import moment, { Moment } from 'moment'

export type FormatFunctionType = (
  date: Moment,
  options: {
    format?: string
  }
) => string
export const ISO8601_FORMAT = 'ISO8601_FORMAT'

class DateFormatter {
  fallbackFormat?: string
  defaultLocale: string

  constructor(defaultLocale: string, fallbackFormat: string | undefined = ISO8601_FORMAT) {
    this.fallbackFormat = fallbackFormat
    this.defaultLocale = defaultLocale
  }

  format: FormatFunctionType = (
    date: Moment,
    options: {
      format?: string
    }
  ) => {
    const format = options.format || this.fallbackFormat
    // TODO IGAPP-399: Uncomment again and use locale instead of hardcoded 'en'
    // const requestedLocale = defaultLocale
    const requestedLocale = 'en'
    const allLocales = moment.locales()
    const locale = allLocales.includes(requestedLocale) ? requestedLocale : this.defaultLocale
    return date.locale(locale).format(format)
  }
}

export default DateFormatter
