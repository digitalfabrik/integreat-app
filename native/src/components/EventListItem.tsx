import * as React from 'react'
import { ReactElement, useCallback, useContext } from 'react'
import styled from 'styled-components/native'

import { EventModel, EVENTS_ROUTE, RouteInformationType } from 'api-client'
import { ThemeType } from 'build-configs'

import EventPlaceholder1 from '../assets/EventPlaceholder1.jpg'
import EventPlaceholder2 from '../assets/EventPlaceholder2.jpg'
import EventPlaceholder3 from '../assets/EventPlaceholder3.jpg'
import DateFormatterContext from '../contexts/DateFormatterContext'
import ListItem from './ListItem'

type PropsType = {
  cityCode: string
  event: EventModel
  language: string
  navigateTo: (arg0: RouteInformationType) => void
  theme: ThemeType
}
const Description = styled.Text`
  color: ${props => props.theme.colors.textColor};
  font-family: ${props => props.theme.fonts.native.contentFontRegular};
`

/**
 * We have three placeholder thumbnails to display when cities don't provide a thumbnail
 * @returns {*} The Placeholder Thumbnail
 */
const getEventPlaceholder = (id: number): number => {
  const placeholders = [EventPlaceholder1, EventPlaceholder2, EventPlaceholder3]
  return placeholders[id % placeholders.length]
}

const EventListItem = ({ event, cityCode, language, navigateTo, theme }: PropsType): ReactElement => {
  const formatter = useContext(DateFormatterContext)
  const navigateToEventInCity = useCallback(() => {
    navigateTo({
      route: EVENTS_ROUTE,
      cityCode,
      languageCode: language,
      cityContentPath: event.path
    })
  }, [navigateTo, cityCode, language, event])
  const thumbnail = event.thumbnail || getEventPlaceholder(event.path.length)
  return (
    <ListItem
      thumbnail={thumbnail}
      title={event.title}
      language={language}
      navigateTo={navigateToEventInCity}
      theme={theme}>
      <Description theme={theme}>{event.date.toFormattedString(formatter)}</Description>
      {event.location.location && <Description theme={theme}>{event.location.location}</Description>}
    </ListItem>
  )
}

export default EventListItem
