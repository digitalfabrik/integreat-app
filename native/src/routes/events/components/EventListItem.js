// @flow

import * as React from 'react'
import { useCallback, useContext } from 'react'
import { EventModel } from 'api-client'
import ListItem from '../../../modules/common/components/ListItem'
import EventPlaceholder1 from '../assets/EventPlaceholder1.jpg'
import EventPlaceholder2 from '../assets/EventPlaceholder2.jpg'
import EventPlaceholder3 from '../assets/EventPlaceholder3.jpg'
import styled from 'styled-components/native'
import type { ThemeType } from 'build-configs/ThemeType'
import DateFormatterContext from '../../../modules/i18n/context/DateFormatterContext'
import { EVENTS_ROUTE } from 'api-client/src/routes'
import type { RouteInformationType } from 'api-client/src/routes/RouteInformationTypes'
import type { StyledComponent } from 'styled-components'

type PropsType = {|
  cityCode: string,
  event: EventModel,
  language: string,
  navigateTo: RouteInformationType => void,
  theme: ThemeType
|}

const Description: StyledComponent<{||}, ThemeType, *> = styled.Text`
  color: ${props => props.theme.colors.textColor};
  font-family: ${props => props.theme.fonts.native.contentFontRegular};
`

/**
 * We have three placeholder thumbnails to display when cities don't provide a thumbnail
 * @returns {*} The Placeholder Thumbnail
 */
const getEventPlaceholder = (id: number): number => {
  const placeholders = [EventPlaceholder1, EventPlaceholder2, EventPlaceholder3]
  return placeholders[id % placeholders.length]
}

const EventListItem = ({ event, cityCode, language, navigateTo, theme }: PropsType) => {
  const formatter = useContext(DateFormatterContext)

  const navigateToEventInCity = useCallback(() => {
    navigateTo({
      route: EVENTS_ROUTE,
      cityCode,
      languageCode: language,
      cityContentPath: event.path
    })
  }, [navigateTo, cityCode, language, event])

  const thumbnail = event.thumbnail || getEventPlaceholder(event.path.length)
  return (
    <ListItem
      thumbnail={thumbnail}
      title={event.title}
      language={language}
      navigateTo={navigateToEventInCity}
      theme={theme}>
      <Description theme={theme}>{event.date.toFormattedString(formatter)}</Description>
      {event.location.location && <Description theme={theme}>{event.location.location}</Description>}
    </ListItem>
  )
}

export default EventListItem
