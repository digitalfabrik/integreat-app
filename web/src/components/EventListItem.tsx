import React, { ReactElement } from 'react'
import { EventModel, DateFormatter } from 'api-client'
import ListItem from './ListItem'
import EventPlaceholder1 from '../assets/EventPlaceholder1.jpg'
import EventPlaceholder2 from '../assets/EventPlaceholder2.jpg'
import EventPlaceholder3 from '../assets/EventPlaceholder3.jpg'
import { textTruncator } from '../utils/stringUtils'

export const NUM_OF_WORDS_ALLOWED = 15

type PropsType = {
  event: EventModel
  formatter: DateFormatter
}

/**
 * We have three placeholder thumbnails to display when cities don't provide a thumbnail
 * @returns {*} The Placeholder Thumbnail
 */
const getEventPlaceholder = (path: string): string => {
  const pseudoId = path.split('').reduce((acc, char) => acc + char.charCodeAt(0), 0)
  const placeholders = [EventPlaceholder1, EventPlaceholder2, EventPlaceholder3]
  return placeholders[pseudoId % placeholders.length]
}

const EventListItem = ({ event, formatter }: PropsType): ReactElement => {
  return (
    <ListItem thumbnail={event.thumbnail || getEventPlaceholder(event.path)} title={event.title} path={event.path}>
      <div>
        <div>{event.date.toFormattedString(formatter)}</div>
        {event.location.location && <div>{event.location.location}</div>}
      </div>
      <div>{textTruncator(event.excerpt, NUM_OF_WORDS_ALLOWED)}</div>
    </ListItem>
  )
}

export default EventListItem
