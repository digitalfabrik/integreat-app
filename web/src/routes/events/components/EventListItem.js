// @flow

import * as React from 'react'
import { EventModel } from 'api-client'
import ListItem from '../../../modules/common/components/ListItem'
import EventPlaceholder1 from '../assets/EventPlaceholder1.jpg'
import EventPlaceholder2 from '../assets/EventPlaceholder2.jpg'
import EventPlaceholder3 from '../assets/EventPlaceholder3.jpg'
import DateFormatter from 'api-client/src/i18n/DateFormatter'

const EXCERPT_LENGTH = 70

type PropsType = {|
  event: EventModel,
  formatter: DateFormatter
|}

const formatExcerpt = (excerpt: string): string => {
  return `${excerpt.slice(0, EXCERPT_LENGTH)}...`
}

/**
 * We have three placeholder thumbnails to display when cities don't provide a thumbnail
 * @returns {*} The Placeholder Thumbnail
 */
const getEventPlaceholder = (path: string): string => {
  const pseudoId = path.split('').reduce((acc, char) => acc + char.charCodeAt(0), 0)
  const placeholders = [EventPlaceholder1, EventPlaceholder2, EventPlaceholder3]
  return placeholders[pseudoId % placeholders.length]
}

const EventListItem = ({
  event,
  formatter
}: PropsType) => {
  return (
    <ListItem thumbnail={event.thumbnail || getEventPlaceholder(event.path)}
              title={event.title}
              path={event.path}>
      <div>
        <div>{event.date.toFormattedString(formatter)}</div>
        {event.location.location && <div>{event.location.location}</div>}
      </div>
      <div>{formatExcerpt(event.excerpt)}</div>
    </ListItem>
  )
}

export default EventListItem
