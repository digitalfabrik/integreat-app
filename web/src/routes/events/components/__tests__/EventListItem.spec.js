// @flow

import React from 'react'
import { EventModel, LocationModel, DateModel } from 'api-client'
import moment from 'moment'
import { shallow } from 'enzyme'
import EventListItem from '../EventListItem'
import DateFormatter from 'api-client/src/i18n/DateFormatter'

describe('EventListItem', () => {
  const language = 'de'

  const event = new EventModel({
    path: '/augsburg/en/events/first_event',
    title: 'first Event',
    availableLanguages: new Map([
      ['de', '/augsburg/de/events/erstes_event'],
      ['ar', '/augsburg/ar/events/erstes_event']
    ]),
    date: new DateModel({
      startDate: moment('2017-11-18T09:30:00.000Z'),
      endDate: moment('2017-11-18T19:30:00.000Z'),
      allDay: true
    }),
    location: new LocationModel({
      name: 'name',
      address: 'address',
      town: 'town',
      postcode: 'postcode',
      longitude: null,
      latitude: null,
      state: 'state',
      region: 'region',
      country: 'country'
    }),
    excerpt: 'very loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong excerpt',
    lastUpdate: moment('2016-01-07 10:36:24'),
    content: 'content',
    thumbnail: 'thumbnail',
    hash: '2fe6283485a93932',
    featuredImage: null
  })

  // TODO IGAPP-399: Reactivate test
  // eslint-disable-next-line jest/no-disabled-tests
  it.skip('should render and match snapshot', () => {
    expect(shallow(<EventListItem event={event} formatter={new DateFormatter(language)} />)).toMatchSnapshot()
  })
})
