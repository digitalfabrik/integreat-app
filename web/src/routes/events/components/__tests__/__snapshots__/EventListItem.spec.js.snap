// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventListItem should render and match snapshot 1`] = `
<ListItem
  isExternalUrl={false}
  path="/augsburg/en/events/first_event"
  thumbnail="thumbnail"
  title="first Event"
>
  <div>
    <div>
      18. November 2017
    </div>
    <div>
      name, address, postcode town
    </div>
  </div>
  <div>
    very loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo...
  </div>
</ListItem>
`;
